### PR TITLE
New contributing.rst file

### PR DESCRIPTION
**Pull request by hauten**
This file includes everything from http://flux-framework.org/docs/contributing/ (which I will delete after this page is published) plus additional text and links to flesh out the contributing guidelines a bit more. There's more to do (see issue #1, which is NOT fully resolved by this PR), but I wanted something we could to start with.

### CODE REVIEW

File: contributing.rst [hunk 1]
```diff
@@ -4,16 +4,37 @@
...
 
-* Fork the project.
-* Clone your fork. ``git clone git@github.com:[username]/flux-framework/[project].git``
+.. _pull-requests:
+
+-------------
+Pull Requests
+-------------
+
+Use a pull request (PR) toward a repository's master branch to propose your contribution. If you are planning significant code changes, or have any questions, you can also open an issue before submitting a PR. To contribute to a Flux Framework project:
```
Comment (garlick) [inline contributing.rst]: Maybe:

> If you are planning significant code changes, are fixing a bug, or have any questions, please open...

Ideally, PR's reference an issue where a problem was reported, or where a new feature was proposed and discussed.  (This is a SHOULD not a MUST in our RFC 1).  So I think the language about opening an issue should be a little stronger but not dictatorial.

File: contributing.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+--------------------
+Developer Guidelines
+--------------------
+
+* Keep the code lean and as simple as possible.
+* Keep the code general and reasonably efficient.
+* Review other issues and PRs to ensure you are not duplicating effort.
+* Update the appropriate `documentation <https://github.com/flux-framework/docs>`_ or open an `issue <https://github.com/flux-framework/docs/issues>`_ to do so.
+
+For more details about C4.1, see LINK TO COME.
```
Comment (SteVwonder) [inline contributing.rst]:
```suggestion
For more details about C4.1, see `RFC 1 <https://github.com/flux-framework/rfc/blob/master/spec_1.adoc>`.
```

**Comment by SteVwonder** [comment]
> I have to suppress the urge to add stuff like "no merge commits!" and "each commit fixes one problem!" and "use subsystem: prefix in commits titles!".

These might be things worth putting in a PR checklist that is used as part of a PR template on github.

**Comment by hauten** [comment]
@SteVwonder as we discussed, you can merge this and I'll create a new branch picking up where I left off & incorporating this feedback. My push/pull situation is resolved properly now.

**Comment by garlick** [comment]
Sounds like this is ready to go so merging.  Thanks!

**Comment by hauten** [comment]
> use subsystem: prefix in commits titles


> > I have to suppress the urge to add stuff like "no merge commits!" and "each commit fixes one problem!" and "use subsystem: prefix in commits titles!".
> 
> These might be things worth putting in a PR checklist that is used as part of a PR template on github.

Adding these to the "Developer Guidelines" section for now. I created issue #11 for a PR template; not sure where in the Docs it would go and what other kinds of things to include, but just want an issue so we don't forget it exists.

**Pull request merged**